state/elasticsearch.yaml

```diff
@@ -163,25 +171,37 @@ Resources:
         DedicatedMasterCount: !If [HasDedicatedMasterNodes, !Ref DedicatedMasterCount, !Ref 'AWS::NoValue']
         DedicatedMasterEnabled: !If [HasDedicatedMasterNodes, true, false]
         DedicatedMasterType: !If [HasDedicatedMasterNodes, !Ref DedicatedMasterType, !Ref 'AWS::NoValue']
-        InstanceCount: !Ref ClusterInstanceCount
+        InstanceCount: !If
```

Comment:
Why did you decide to do it like that? Why not just use the `InstanceCount` in any case? Wouldn't that be less complex?

Reply:
I always try to automatically set the correct values for dependent values and the count of instances must be at least the number of availability zones. I contemplated setting the default value to 0 and only then use the number of availability zones as instance count. But I was not sure if this will cause an issue if someone updates the template and is not aware, that the default value has changed.

Reply:
I think we should add one parameter `AvailabilityZoneCount` that can be either 2 or 3.

If InstanceCount = 1: we ignore AvailabilityZoneCount and use subnet A.
If InstanceCount > 1 && AvailabilityZoneCount = 2: we use subnet A, and B.
If InstanceCount > 1 && AvailabilityZoneCount = 3: we use subnetA, B, and C.

Would that make sense?